fedimint-core/src/encoding/mod.rs

```diff
@@ -143,7 +144,27 @@ impl DecodeError {
     }
 }
 
-macro_rules! impl_encode_decode_num {
+pub use lightning::util::ser::BigSize;
+
+impl Encodable for BigSize {
+    fn consensus_encode<W: std::io::Write>(&self, writer: &mut W) -> Result<usize, std::io::Error> {
+        let mut writer = CountWrite::from(writer);
+        self.write(&mut writer)?;
+        Ok(usize::try_from(writer.count()).expect("can't overflow"))
+    }
+}
+
+impl Decodable for BigSize {
+    fn consensus_decode<R: std::io::Read>(
+        r: &mut R,
+        _modules: &ModuleDecoderRegistry,
+    ) -> Result<Self, DecodeError> {
+        BigSize::read(r)
+            .map_err(|e| DecodeError::new_custom(anyhow::anyhow!("BigSize decoding error: {e}")))
```

Comment:
Always use `{:?}` for errors to preserve the backtrace
```suggestion
            .map_err(|e| DecodeError::new_custom(anyhow::anyhow!("BigSize decoding error: {e:?}")))
```

Reply:
> Always use {:?} for errors to preserve the backtrace

Ohh... I didn't know that. Is this documented anywhere? Could we have a lint for it?

Reply:
Not sure if it's documented, but it's a fact that I've learned after a lot of pain. I don't think clippy supports that warning and manually implementing such linter is tricky.